drivers/interrupt_controller/intc_qingke_pfic.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2023 Chen Xingyu <hi@xingrz.me>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT wch_qingke_pfic
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/arch/riscv/irq.h>
+#include <zephyr/device.h>
+
+#define PFIC_BASE DT_INST_REG_ADDR(0)
+
+#define PFIC_ISR(n)      (PFIC_BASE + 0x000 + 0x4 * n)
+#define PFIC_IPR(n)      (PFIC_BASE + 0x020 + 0x4 * n)
+#define PFIC_ITHRESDR    (PFIC_BASE + 0x040)
+#define PFIC_VTFBADDRR   (PFIC_BASE + 0x044)
+#define PFIC_CFGR        (PFIC_BASE + 0x048)
+#define PFIC_GISR        (PFIC_BASE + 0x04C)
+#define PFIC_IDCFGR      (PFIC_BASE + 0x050)
+#define PFIC_VTFADDRR(n) (PFIC_BASE + 0x060 + 0x4 * n)
+#define PFIC_IENR(n)     (PFIC_BASE + 0x100 + 0x4 * n)
+#define PFIC_IRER(n)     (PFIC_BASE + 0x180 + 0x4 * n)
+#define PFIC_IPSR(n)     (PFIC_BASE + 0x200 + 0x4 * n)
+#define PFIC_IPRR(n)     (PFIC_BASE + 0x280 + 0x4 * n)
+#define PFIC_IACTR(n)    (PFIC_BASE + 0x300 + 0x4 * n)
+#define PFIC_IPRIOR(n)   (PFIC_BASE + 0x400 + n)
+#define PFIC_SCTLR       (PFIC_BASE + 0xD10)
+
+/* PFIC_ITHRESDR Register */
+#define PFIC_ITHRESDR_THRESHOLD(n) ((n & BIT_MASK(4)) << 4)
+
+#define PFIC_IRQN_GROUP(irqn) (irqn >> 5)
+#define PFIC_IRQN_SHIFT(irqn) (irqn & BIT_MASK(5))
+
+void riscv_clic_irq_enable(unsigned int irq)
```

Comment:
QingKe PFIC is not "CLIC".

Besides with this change, 
https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/riscv/riscv-privileged/common/soc_common_irq.c#L18-L39

should be better to modify for generalizing.
At this point, I want to hear @carlocaione opinions.

Reply:
Hi @soburi , @xingrz ,

We also faced this recently: https://github.com/zephyrproject-rtos/zephyr/pull/62606
Looks like we indeed need to come-up with some common API to be used with our custom interrupt controllers.
https://github.com/zephyrproject-rtos/zephyr/pull/63214 is targeting this API